seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxdbSourceReader.java

```diff
@@ -151,4 +171,49 @@ private void read(InfluxDBSourceSplit split, Collector<SeaTunnelRow> output) {
             }
         }
     }
+
+    private void readByChunkSize(InfluxDBSourceSplit split, Collector<SeaTunnelRow> output) {
+        influxdb.query(
+                new Query(split.getQuery(), config.getDatabase()),
```

Comment:
Use `CountDownLatch` is not a good idea. You can update the code like this: (Then method `readByChunkSize` will become a sync methods and will not return until the reading thread is completed )

```
final CompletableFuture<Void> queryCompleteFeature = new CompletableFuture<>();

influxdb.query(
                new Query(split.getQuery(), config.getDatabase()),
config.getChunkSize(),
                (cancellable, queryResult) -> {
                    if (cancellable.isCanceled()) {
                        log.info("this chunk reader influxDB is canceled");
                        queryCompleteFeature.complete();
                        return;
                    }
                    if (queryResult.hasError()) {
                        log.error(
                                "this chunk reader influxDB result has error [{}]",
                                queryResult.getError());
                        queryCompleteFeature.completeExceptionally(new InfluxdbConnectorException)
                        return;
                    }
                    for (QueryResult.Result result : queryResult.getResults()) {
                        List<QueryResult.Series> serieList = result.getSeries();
                        if (CollectionUtils.isNotEmpty(serieList)) {
                            for (QueryResult.Series series : serieList) {
                                for (List<Object> values : series.getValues()) {
                                    SeaTunnelRow row =
                                            InfluxDBRowConverter.convert(
                                                    values, seaTunnelRowType, columnsIndexList);
                                    output.collect(row);
                                }
                            }
                        } else {
                            log.info("this chunk reader influxDB series is empty");
                        }
                    }
                },
                () -> {
                    log.info("this chunk reader influxDB complete");
                    queryCompleteFeature.complete();
                },
                throwable -> {
                    log.error(
                            "this chunk reader influxDB result has error [{}]",
                            throwable.getMessage());
                    queryCompleteFeature.completeExceptionally(new InfluxdbConnectorException(throwable))
                });
    }

   queryCompleteFeature.get();
```

Reply:
Yes, I've already tried to switch to CompletableFuture yesterday, but I haven't got it working yet due to limited time. I know CountDownLatch is not a good choice.

Reply:
But merely changing this method to a synchronous one is not enough, as it occupies the lock in the pollNext method.

Reply:
> But merely changing this method to a synchronous one is not enough, as it occupies the lock in the pollNext method.

It is normal to occupy a lock, and we must ensure that the lock(`output.getCheckpointLock()`) cannot be released until a split read is completed